mne/io/curry/curry.py

```diff
@@ -0,0 +1,244 @@
+            file_extension = '.cdt.ceo'
+        else:
+            curry_events = None
+
+    if curry_events is not None:
+
+        save_events = False
+        with open(fname_base + file_extension) as fid:
+            for line in fid:
+
+                if "NUMBER_LIST END_LIST" in line:
+                    save_events = False
+
+                if save_events:
+                    # print(line)
+                    curry_events.append(line.split("\t"))
+
+                if "NUMBER_LIST START_LIST" in line:
+                    save_events = True
+
+        curry_events = np.array(curry_events, dtype=int)
+
+    # TODO: This returns events in a curry specific format. This might be reformatted to fit mne
+    return curry_events
+
+
+def _check_curry_file(full_fname):
+    """
+    Check if all neccessary files exist and return the path without extension
+     and its CURRY version
+     """
+
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = full_fname.split(".", maxsplit=1)
+
+    if 'cdt' in ext:
+        curry_vers = 8
+        for check_ext in [".cdt", ".cdt.dpa"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    else:
+        curry_vers = 7
+        for check_ext in [".dap", ".dat", ".rs3"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    return fname_base, curry_vers
+
+
+def _read_curry_info(fname_base, curry_vers):
+    #####################################
+    # read parameters from the param file
+
+    if curry_vers == 7:
+        file_extension = '.dap'
+    else:
+        file_extension = '.cdt.dpa'
+
+    var_names = ['NumSamples', 'NumChannels', 'NumTrials', 'SampleFreqHz',
+                 'TriggerOffsetUsec', 'DataFormat', 'SampleTimeUsec',
+                 'NUM_SAMPLES', 'NUM_CHANNELS', 'NUM_TRIALS', 'SAMPLE_FREQ_HZ',
+                 'TRIGGER_OFFSET_USEC', 'DATA_FORMAT', 'SAMPLE_TIME_USEC']
+
+    param_dict = dict()
+    with open(fname_base + file_extension) as fid:
+        for line in fid:
+            if any(var_name in line for var_name in var_names):
+                key, val = line.replace(" ", "").replace("\n", "").split("=")
+                param_dict[key] = val
+
+    if not len(param_dict) == 7:
+        raise KeyError("Some variables cannot be found in the parameter file.")
```

Comment:
This is a bit unspecific hm? You could loop over the expected values here and raise an error pointing out what exactly is missing.

Do we need everything? Can it still work if n_trials is missing? Aren't time_step and sfreq redundant?

Reply:
(I.e. you could loop over the first 7 entries in var_names)

Reply:
Yes, I will of course remove the unneeded variables before finalizing it, but for now, i wanted to keep them in since for example data_format is used to decide between ASCII coding later...
The other might be useful to double check. but yeah, we might eventally get rid of n_ch, n_trials and sfreq